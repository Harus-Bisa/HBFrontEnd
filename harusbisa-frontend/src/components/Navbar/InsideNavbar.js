import React from "react";
import {connect} from "react-redux";
import {
    Collapse,
    Navbar,
    NavbarToggler,
    NavbarBrand,
    Nav,
    NavItem,
    NavLink,
    UncontrolledDropdown,
    DropdownToggle,
    DropdownMenu,
    DropdownItem } from 'reactstrap';

function mapStateToProps(state){
    return{
        firstName: state.firstName, 
        lastName: state.lastName,
        courses: state.courses
    }
}
function InsideNavbar(props){
    var [isOpen, setIsOpen] = React.useState(false);
    const toggle = () => setIsOpen(!isOpen);
    const makeCourses = ()  => {
        var courses = []
        for (let i=0; i<props.courses.length; i++){
            courses.push(
                <DropdownItem key={i}><NavLink href={"/student/dashboard/"+props.courses[i]._id}>{props.courses[i].course_name}</NavLink></DropdownItem>
            )
        }
        return courses;

    }
    return(
        <div style={{background:'#f4f4f4'}}>
            <div className="container">
                <div className="d-none d-md-block">
                    <div className="row justify-content-center">
                        <div className="col-4">
                            <p>Dropdown 1</p>
                        </div>
                        <div className="col-4">
                            Logo
                        </div>
                        <div className="col-4">
                            <p>{props.firstName} {props.lastName}</p>
                            <button onClick={props.logout}>Logout</button>
                        </div>
                    </div>
                </div>
                <div className="d-sm-none">
                    <Navbar expand="md" color="faded" light>
                        <NavbarBrand href="/">Harus Bisa</NavbarBrand>
                        <button>+</button>
                        <NavbarToggler onClick={toggle} className="mr-2"/>
                        <Collapse isOpen={isOpen} navbar>
                            <Nav navbar>
                                <UncontrolledDropdown nav inNavbar>
                                    <DropdownToggle nav caret>
                                        Kelas
                                    </DropdownToggle>
                                    <DropdownMenu>
                                        {makeCourses()}
                                    </DropdownMenu>
                                </UncontrolledDropdown>
                                <UncontrolledDropdown nav inNavbar>
                                    <DropdownToggle nav caret>
                                        {props.firstName} {props.lastName}
                                    </DropdownToggle>
                                    <DropdownMenu>
                                        <DropdownItem>Setting</DropdownItem>
                                        <DropdownItem>Bantuan</DropdownItem>
                                    </DropdownMenu>
                                </UncontrolledDropdown>
                                <NavItem onClick={props.logout}>
                                    Log off
                                </NavItem>
                            </Nav>
                        </Collapse>
                    </Navbar>
                </div>
            </div>
        </div>
    )
}

export default connect(mapStateToProps)(InsideNavbar);
