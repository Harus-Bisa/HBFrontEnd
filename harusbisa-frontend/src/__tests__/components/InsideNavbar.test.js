import React from "react";
import InsideNavbar from "../../components/Navbar/InsideNavbar";
import {Provider} from "react-redux";
import {mount} from "enzyme";
import configureMockStore from "redux-mock-store";
import '../../test-config';
import {BrowserRouter as Router} from "react-router-dom";
import thunk from 'redux-thunk';

const mockStore = configureMockStore([thunk]);
describe("InsideNavbar component", () =>{
    let initialState;
    let insideNavbar;
    let store;

    beforeEach(() =>{
        initialState = {
            courses:[{
                "_id": "5db492c349a67b33b8d0a2a2",
                "course_name": "Calculus I",
                "course_description": "The basic of calculus (derivatives).",
                "start_term": "August 2019",
                "end_term": "December 2019",
                "number_of_lectures": 5,
                "instructor": "John Doe"
              }],
            loading: false
        }
        store = mockStore(initialState);
        store.dispatch = jest.fn()
        insideNavbar = mount(<Router><Provider store={store}><InsideNavbar/></Provider></Router>)
    })

    it("Matches snapshot", () =>{
        expect(insideNavbar.html()).toMatchSnapshot();
    })
    
    it("Renders the small navbar log off button", () =>{
        var smallLogoffButton = insideNavbar.find("#small-logoff").at(0);
        expect(store.dispatch).not.toHaveBeenCalled()
        smallLogoffButton.simulate('click')
        expect(store.dispatch).toHaveBeenCalled()
        expect(store.dispatch).toHaveBeenCalledTimes(1)
       
    })

    it("Renders the big navbar log off button", () =>{
        var bigLogoffButton = insideNavbar.find("#big-logoff").at(0);
        expect(store.dispatch).not.toHaveBeenCalled()
        bigLogoffButton.simulate('click')
        expect(store.dispatch).toHaveBeenCalled()
        expect(store.dispatch).toHaveBeenCalledTimes(1)
       
    })

})