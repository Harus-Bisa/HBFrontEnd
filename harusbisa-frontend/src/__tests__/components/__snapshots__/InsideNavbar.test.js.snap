// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InsideNavbar component Matches snapshot 1`] = `"<div style=\\"background: rgb(244, 244, 244); position: sticky; top: 0px; z-index: 999;\\"><div class=\\"container\\"><div class=\\"d-none d-md-block\\"><nav class=\\"navbar navbar-expand-md navbar-light\\"><ul style=\\"width: 100%; justify-content: space-between;\\" class=\\"navbar-nav\\"><li class=\\"dropdown nav-item\\"><a aria-haspopup=\\"true\\" href=\\"#\\" class=\\"dropdown-toggle nav-link\\" aria-expanded=\\"false\\">Kelas</a><div tabindex=\\"-1\\" role=\\"menu\\" aria-hidden=\\"true\\" class=\\"dropdown-menu\\"><button type=\\"button\\" tabindex=\\"0\\" role=\\"menuitem\\" class=\\"dropdown-item\\"><a href=\\"/student/dashboard/5db492c349a67b33b8d0a2a2\\" class=\\"nav-link\\">Calculus I</a></button></div></li><a href=\\"/\\" class=\\"navbar-brand\\"><div class=\\"full \\" style=\\"height: 2rem;\\"><img alt=\\"logo\\" class=\\"hvr-bob\\" src=\\"Harus_Bisa-07-Full-Blk-Trans-NoPad_1.png\\"><img alt=\\"logo\\" src=\\"Harus_Bisa-07-Full-Blk-Trans-NoPad_2.png\\"></div></a><li class=\\"dropdown nav-item\\"><a aria-haspopup=\\"true\\" href=\\"#\\" class=\\"dropdown-toggle nav-link\\" aria-expanded=\\"false\\"> </a><div tabindex=\\"-1\\" role=\\"menu\\" aria-hidden=\\"true\\" class=\\"dropdown-menu\\"><button type=\\"button\\" tabindex=\\"0\\" role=\\"menuitem\\" class=\\"dropdown-item\\">Notifikasi</button><div tabindex=\\"-1\\" class=\\"dropdown-divider\\"></div><button type=\\"button\\" tabindex=\\"0\\" role=\\"menuitem\\" class=\\"dropdown-item\\">Setting</button><button type=\\"button\\" tabindex=\\"0\\" role=\\"menuitem\\" class=\\"dropdown-item\\">Bantuan</button><button type=\\"button\\" id=\\"big-logoff\\" tabindex=\\"0\\" role=\\"menuitem\\" class=\\"dropdown-item\\">Log off</button></div></li></ul></nav></div><div class=\\"d-md-none\\"><nav class=\\"navbar navbar-expand-md navbar-light bg-faded\\"><a href=\\"/\\" class=\\"navbar-brand\\"><div class=\\"full \\" style=\\"height: 2rem;\\"><img alt=\\"logo\\" class=\\"hvr-bob\\" src=\\"Harus_Bisa-07-Full-Blk-Trans-NoPad_1.png\\"><img alt=\\"logo\\" src=\\"Harus_Bisa-07-Full-Blk-Trans-NoPad_2.png\\"></div></a><div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\">+</span><span class=\\"MuiTouchRipple-root\\"></span></button></div><button aria-label=\\"Toggle navigation\\" type=\\"button\\" class=\\"mr-2 navbar-toggler\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"collapse navbar-collapse\\" aria-expanded=\\"false\\"><ul class=\\"navbar-nav\\"><li class=\\"dropdown nav-item\\"><a aria-haspopup=\\"true\\" href=\\"#\\" class=\\"dropdown-toggle nav-link\\" aria-expanded=\\"false\\">Kelas</a><div tabindex=\\"-1\\" role=\\"menu\\" aria-hidden=\\"true\\" class=\\"dropdown-menu\\"><button type=\\"button\\" tabindex=\\"0\\" role=\\"menuitem\\" class=\\"dropdown-item\\"><a href=\\"/student/dashboard/5db492c349a67b33b8d0a2a2\\" class=\\"nav-link\\">Calculus I</a></button></div></li><li class=\\"dropdown nav-item\\"><a aria-haspopup=\\"true\\" href=\\"#\\" class=\\"dropdown-toggle nav-link\\" aria-expanded=\\"false\\"> </a><div tabindex=\\"-1\\" role=\\"menu\\" aria-hidden=\\"true\\" class=\\"dropdown-menu\\"><button type=\\"button\\" tabindex=\\"0\\" role=\\"menuitem\\" class=\\"dropdown-item\\">Setting</button><button type=\\"button\\" tabindex=\\"0\\" role=\\"menuitem\\" class=\\"dropdown-item\\">Bantuan</button></div></li><li id=\\"small-logoff\\" class=\\"nav-item\\">Log off</li></ul></div></nav></div></div></div>"`;
